external/storm-kafka/src/test/storm/kafka/KafkaUtilsTest.java

```diff
@@ -168,7 +168,7 @@ public void generateTuplesWithValueSchemeAndKeyValueMessage() {
     }
 
     private ByteBufferMessageSet getLastMessage() {
-        long offsetOfLastMessage = KafkaUtils.getOffset(simpleConsumer, config.topic, 0, OffsetRequest.LatestTime()) - 1;
+        long offsetOfLastMessage = KafkaUtils.getOffset(simpleConsumer, config.topic, 0, OffsetRequest.LatestTime());
```

Comment:
To address this concern, I just updated the producer in KafkaUtilsTest with new kafka Producer API, and use "send().get()" to do synchronous sending. Then the offset uncertainty can be fixed and we no longer need to change that offset line in the test.